lsp-mode.el

```diff
@@ -7192,19 +7192,32 @@ nil."
 
 
 ;; Download URL handling
-(cl-defun lsp-download-install (callback error-callback &key url store-path &allow-other-keys)
-  (let ((url (lsp-resolve-value url))
-        (store-path (lsp-resolve-value store-path)))
+(cl-defun lsp-download-install (callback error-callback &key url store-path decompress &allow-other-keys)
+  (let* ((url (lsp-resolve-value url))
+         (store-path (lsp-resolve-value store-path))
+         ;; (decompress (lsp-resolve-value decompress))
+         (download-path
+          (pcase decompress
+           (`gzip (concat store-path ".gz"))
+           (`zip (concat store-path ".zip"))
+           (`nil store-path)
+           (_ (error ":decompress must be `gzip', `zip' or `nil'")))))
     (make-thread
      (lambda ()
        (condition-case err
            (progn
-             (when (f-exists? store-path)
-               (f-delete store-path))
-             (lsp--info "Starting to download %s to %s..." url store-path)
-             (mkdir (f-parent store-path) t)
-             (url-copy-file url store-path)
-             (lsp--info "Finished downloading %s..." store-path)
+             (when (f-exists? download-path)
+               (f-delete download-path))
+             (lsp--info "Starting to download %s to %s..." url download-path)
+             (mkdir (f-parent download-path) t)
+             (url-copy-file url download-path)
+             (lsp--info "Finished downloading %s..." download-path)
+             (when decompress
+               (lsp--info "Decompressing %s..." download-path)
+               (pcase decompress
+                 (`gzip (lsp-gunzip download-path))
```

Comment:
This can be done asynchronously. `lsp-gunzip-async` could get CALLBACK and ERROR-CALLBACK as parameters, and call `lsp-start-process-async` instead. An async version of `lsp-unzip` should be added as well.

Reply:
@bubba you may go without this

Reply:
I can do that one afterwards.